src/SSHLibrary/javaclient.py

```diff
@@ -176,6 +181,30 @@ def _read_from_stream(self, stream):
             line = reader.readLine()
         return result
 
-    def _execute(self):
+    def _execute(self, sudo=False, sudo_password=None):
         command = self._command.decode(self._encoding)
         self._shell.execCommand(command)
+
+
+class RemoteSudoCommand(RemoteCommand):
+    def __init__(self, command, encoding):
+        super(RemoteSudoCommand, self).__init__(command, encoding)
+        self._stdout = ''
+
+    def read_outputs(self):
+        stderr = self._read_from_stream(self._shell.getStderr())
+        rc = self._shell.getExitStatus() or 0
+        self._shell.close()
+        return self._stdout, stderr, rc
+
+    def _execute(self, sudo=False, sudo_password=None):
+        command = 'sudo ' + self._command.decode(self._encoding)
+        self._shell.execCommand(command)
+        if sudo_password is not None:
+            stdin = self._shell.getStdin()
+            try:
+                while True:
+                    self.send_password(sudo_password, stdin)
+            except IOException:
+                pass
+            self._stdout = self._read_from_stream(self._shell.getStdout())
```

Comment:
1. I find it a bit confusing that on Java side there's separate `RemoteSudoCommand` but not on Python side. Couldn't possible differences between normal and sudo usage be handled without a separate class also here? Would dispatching to `_execute` and `_execute_with_sudo` in `AbstractCommand.run_in` like I proposed in an earlier comment help?

2. There seems to be a bug that if sudo is used but password is not provided, stdout is not read at all. This obviously needs to be fixed regardless is architecture changed or not. Additional test for sudo usage without password needs to be added as well.

3. The `while True` loop inside `try/except IOException` looks really strange. Could you clarify why it's needed? Why to write password multiple times in genera?

Reply:
2. That bug ought to be fixed now. For the test you propose I think a new test user is needed (one that is set up not to require password on sudo). That means that acceptance test rst file needs to be updated and Travis. Or do you have any idea how to test this without a new user?

3. It was needed to close the shell when the password was incorrect (tried to make it similar on how it was on python side). But, I've completly changed the code on both sides now, so the password is inserted only once. And afterwards close the shell if necessary.